src/lib.rs

```diff
@@ -2,269 +2,186 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-/*! The library provide tools for handling compressed and archive files
-
-# Examples
-```no_run
-use compress_tools;
-
-let dir = tempfile::tempdir().unwrap();
-compress_tools::uncompress("tests/fixtures/tree.tar.gz", dir.path(), compress_tools::Kind::TarGZip).unwrap();
-```
-*/
-
-mod uncompress;
-
-pub use uncompress::{Error, Result};
```

Comment:
Since we don't export multiple errors in this crate, we should keep the module level `Error` and `Result` reexport. So users can write `compress_tools::Error` instead of the more redundant `compress_tools::error::Error`.